### PR TITLE
Use Drift and SQLite for data storage

### LINKED ISSUE #13: Store some user data locally
We'll need to store some of the user's data in a local database on the device.

The minimum version of this for a fully-functioning Zulip client would be:
* Authentication credentials for each of the user's accounts (needed for #11)
* Some other per-account data: the state of registering for push notifications; the server's Zulip version and feature level; the user ID
* User settings that are global to the device (like dark mode vs. light mode), or that otherwise don't come from the server

In particular, that covers everything the existing RN-based app stores and doesn't discard when you switch between accounts.

A future version would also store the full server data: other users with their names and avatars, streams with their names and subscription colors, recent/unread messages, and so on. But we can go a long way without that — the RN-based app has never made great use of this anyway. So we'll file a separate issue for that later.

The scope of this issue is to have local storage in some reasonable form that we think could be good to use in a production app, and to have some data we store there that can change and is reflected in the UI. We'll then add particular pieces of data to it as needed.

### LINKED ISSUE #34: Store login credentials on device
This will be a key followup to #11 to make it a reasonable user experience. For that reason it'll be our first example of #13.

### PR DESCRIPTION
**Pull request by gnprice**
Fixes #13.
Fixes #34.

---
originally:

Toward #13.

This installs Drift as a dependency, sets up a schema for an accounts table, and performs a migration on that table. It also adds tests for the database schema and the migration, to let us see how those look.

I won't want to merge something like this until I've tested it more end-to-end, which will require some other work including #20 and #21. But I think the branch is otherwise ready, apart from a couple of polish issues.

### CODE REVIEW

**Comment by gnprice** [comment]
I've rebased this a couple of times as the related work has progressed, and made some further revisions earlier this week. Just pushed the current version.

This still needs end-to-end testing before I'd want to merge it. That's a lot closer now that #11 is done, and #55. The end-to-end testing will basically consist of implementing #34, storing login credentials.

**Comment by gnprice** [comment]
OK! This is now tested end to end, and I think it's complete. Please take a look.

**Review by chrisbobbe — APPROVED** [review]
Exciting! 🎉

See one comment below, then please merge at will.

**Comment by gnprice** [comment]
Thanks for the review! Done.

File: lib/model/database.dart [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  Column<String> get zulipMergeBase => text().nullable()();
+  Column<int>    get zulipFeatureLevel => integer()();
+
+  Column<String> get ackedPushToken => text().nullable()();
+
+  @override
+  List<Set<Column<Object>>> get uniqueKeys => [
+    {realmUrl, userId},
+    {realmUrl, email},
+  ];
```
[the remark this answers is not in the document]
Comment (gnprice) [reply lib/model/database.dart:29]: Good thought. I added one TODO(log) to log database errors (which we should do in a general way down in the Drift layer), and also a TODO(#35) in lib/widgets/login.dart to give the user feedback if the account is a duplicate.

**Pull request merged**